### ISSUE TITLE
Geany on Windows: Mixture between German and English

### ISSUE THREAD

**Issue by rovf** [geany/geany#1919]
I'm using Geany 1.30.1 on (German) Windows 7. 

What puzzles me is, that the Geany user interface shows a mixture of English and German. The Menu (File, Edit, Search, ...) is completely in English, but the toolbar below is mostly in German, but with English interspersed (Saying "Neu", "öffnen", "Speichern", "Save All", "Zurücksetzen" (whatever this means), "Schließen", "Zurück", "Vor", "Compile", "Build", "Ausführen").

I understand that Geany uses GTK+ to present itself, and that it is a bit difficult to persuade a GTK+ Application to use a certain language. I do not understand, how I can get such a mixture of languages though.

**Comment by elextr** [comment]
> Zurücksetzen

Its a translation for "Reset".

Not sure why the Compile and Build don't translate, they have translations in the po.de file, [eg](https://github.com/geany/geany/blob/3f20ad363a38e9f6b08877d5ade1290d7c69e670/po/de.po#L3785)

Absolutely no idea why the menu isn't in German, @eht16 how German is your windows Geany? :)

**Comment by rovf** [comment]
On Tue, Aug 7, 2018, at 22:21, elextr wrote:
> There is not a selection on install, GTK uses the LANG environment 
> variable, part of the locale.  So to run an application in another 
> language the command (on *ix) `LANG=en geany` is needed.  Not sure how 
> you set environments on windows.

You *can* set environment variables on windows too, though the "canonical" way to do it is to set it in a system-wide manner, so that any application sees the same environment, which we don't want here. To set an environment for a single application, the only way I am aware of is to create a batch file to start this application, and this batch file would implicitly create a command window. Possible, but awkward.

In any case, I would be surprised if LANG is the only parameter which influences the interface language of Geany, because the LANG variable is rarely used in Windows, and Windows has its own way to supply the default language.

The typical way to set a language in an "application specific" way - at least on Windows - is, that the application itself provides this choice. Many applications do this at the time of installation, and some allow to change the interface language "on the fly" while using the program, using a setting in the preferences. If I remember right, I did select "English" at installation time, but I don't know whether this applies only to the installation routine, or also to the installed program. At least this would explain why the vast majority of interface elements is correctly displayed in English.

Ronald

**Comment by b4n** [comment]
A workaround could be (re)moving Geany's translation files so it uses the untranslated (US English) version.

**Comment by rovf** [comment]
On Wed, Aug 8, 2018, at 10:02, Colomban Wendling wrote:
> A workaround could be (re)moving Geany's translation files so it uses 
> the untranslated (US English) version.

Yes I could do this ... but your suggestion makes me recognize that I didn't state the purpose of my posting in a clear way:

It is not that the spurious German texts bother be. I understand them perfectly well, and I don't mind living with a bi-language-Geany. 

The reason why I posted this, is, because the language mixture might be caused by an error in Geany, and the developers of Geany should at least be aware of this effect, and *if* they want to research on this, they can contact me for further information if necessary.

Ronald

**Comment by b4n** [comment]
@rovf oh, OK :)

Reading in more details what you see, is it possible that you actually don't have Geany's translation files, but GTK is installed with its own?  That might explain it, with you seeing strings from GTK translated, and strings from Geany untranslated (because you e.g. didn't install the translations).
I'm not sure, but that sounds plausible.

**Comment by rovf** [comment]
On Wed, Aug 8, 2018, at 15:25, Colomban Wendling wrote:
> @rovf oh, OK :)
> 
> Reading in more details what you see, is it possible that you actually 
> don't have Geany's translation files, but GTK is installed with its own?  
> That might explain it, with you seeing strings from GTK translated, and 
> strings from Geany untranslated (because you e.g. didn't install the 
> translations).

I do use gtk2_prefs.exe to set the appearance of all my GTK applications. However, this program allows only to configure colouring schema and font properties. There is nothing which looks like a language set up.

What translation file do I need to have Geany display all menus in English?

Ronald

**Comment by elextr** [comment]
> What translation file do I need to have Geany display all menus in English?

None, Geany is written in English.  What you need to find is a way of stopping GTK from translating stuff when you don't want it.

**Comment by codebrainz** [comment]
I'm pretty sure the `LANG` variable is the way to do it for both Geany and GTK+ (gettext) on Windows. It can be [set for a particular shortcut](https://netlicensing.io/blog/2012/06/15/set-environment-variables-in-windows-shortcut/) which avoids the need to switch all programs using gettext into the same language system-wide.

**Comment by rovf** [comment]
On Sat, Aug 11, 2018, at 00:43, Matthew Brush wrote:
> I'm pretty sure the `LANG` variable is the way to do it for both Geany 
> and GTK+ (gettext) on Windows. It can be [set for a particular shortcut]
> (https://netlicensing.io/blog/2012/06/15/set-environment-variables-in-windows-shortcut/) 
> which avoids the need to switch all programs using gettext into the same 
> language system-wide.

You were perfectly right! If I set up the shortcut in the way you have proposed, the menus are displayed uniformly in English!

Funny side note: I found in this particular case, that it is sufficient to set the LANG variable to ANY value in order to get the menus in English. Even if I set it to de, fr or ja, I get English menus throughout. I guess that there are simply no other language packs installed for Geany, and since the LANG variable is set, Geany doesn't bother GTK with doing helping in the translation.

Ronald
